packages/loot-core/src/server/filters/app.ts

```diff
@@ -0,0 +1,182 @@
+
+  filters.map(filter => {
+    if (
+      !fCondCheck &&
+      //If conditions.length equals 1 then ignore conditionsOp
+      (conditions.length === 1 ? true : filter.conditionsOp === conditionsOp) &&
+      !filter.tombstone &&
+      filter.conditions.length === conditions.length
+    ) {
+      fCondCheck = false;
+      conditions.map((cond, i) => {
+        condCheck[i] =
+          filter.conditions.filter(fcond => {
+            return (
+              cond.value === fcond.value &&
+              cond.op === fcond.op &&
+              cond.field === fcond.field
+            );
+          }).length > 0;
+        fCondCheck = (i === 0 ? true : fCondCheck) && condCheck[i];
+        return true;
+      });
+      fCondFound = fCondCheck && condCheck[conditions.length - 1] && filter;
+    }
+    return true;
+  });
+
+  condCheck = [];
+
+  if (!newItem) {
+    return fCondFound
+      ? fCondFound.id !== item.id
+        ? fCondFound.name
+        : false
+      : false;
+  }
+  return fCondFound ? fCondFound.name : false;
+}
+
+async function createFilter(filter) {
+  let filterId = uuid.v4Sync();
+  let item = {
+    id: filterId,
+    conditions: filter.state.conditions,
+    conditionsOp: filter.state.conditionsOp,
+    name: filter.state.name,
+  };
+
+  if (item.name) {
+    if (await filterNameExists(item.name, item.id, true)) {
+      throw new Error('There is already a filter named ' + item.name);
+    }
+  } else {
+    throw new Error('Filter name is required');
+  }
+
+  if (item.conditions.length > 0) {
+    let condExists = conditionExists(item, filter.filters, true);
+    if (condExists) {
+      throw new Error(
+        'Duplicate filter warning: conditions already exist. Filter name: ' +
+          condExists,
+      );
+    }
+  } else {
+    throw new Error('Conditions are required');
+  }
+
+  // Create the filter here based on the info
+  await db.insertWithSchema('transaction_filters', filterModel.fromJS(item));
+
+  return filterId;
+}
+
+async function updateFilter(filter) {
+  let item = {
+    id: filter.state.id,
+    conditions: filter.state.conditions,
+    conditionsOp: filter.state.conditionsOp,
+    name: filter.state.name,
+  };
+  if (item.name) {
+    if (await filterNameExists(item.name, item.id, false)) {
+      throw new Error('There is already a filter named ' + item.name);
+    }
+  } else {
+    throw new Error('Filter name is required');
+  }
+
+  if (item.conditions.length > 0) {
+    let condExists = conditionExists(item, filter.filters, false);
+    if (condExists) {
+      throw new Error(
+        'Duplicate filter warning: conditions already exist. Filter name: ' +
+          condExists,
+      );
+    }
+  } else {
+    throw new Error('Conditions are required');
+  }
+
+  await db.updateWithSchema('transaction_filters', filterModel.fromJS(item));
+}
+
+async function deleteFilter({ id }) {
+  await batchMessages(async () => {
+    await db.delete_('transaction_filters', id);
+  });
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I don't think the `batchMessages` wrapping is necessary here since we're doing only a single deletion.